src/huggingface_hub/hf_api.py

```diff
@@ -742,19 +742,19 @@ def create_repo(
             raise ValueError("Invalid repo type")
 
         json = {"name": name, "organization": organization, "private": private}
-        if repo_type is not None:
+        if repo_type is not None and repo_type != "model":
             json["type"] = repo_type
-            if repo_type == "space":
-                if space_sdk is None:
-                    raise ValueError(
-                        "No space_sdk provided. `create_repo` expects space_sdk to be one of "
-                        f"{SPACES_SDK_TYPES} when repo_type is 'space'`"
-                    )
-                if space_sdk not in SPACES_SDK_TYPES:
-                    raise ValueError(
-                        f"Invalid space_sdk. Please choose one of {SPACES_SDK_TYPES}."
-                    )
-                json["sdk"] = space_sdk
+        elif repo_type == "space":
```

Comment:
Actually shouldn't this be an `if` rather than `elif` given that if `repo_type == "space"` would have entered the previous branch of the if statement? 

Would be nice to add a test for that specific behavior to ensure it's not broken.

Thanks @julien-c for raising!